macros/src/hidden.rs

```diff
@@ -0,0 +1,105 @@
+use crate::AttributedThing;
+use quote::{quote, ToTokens};
+use syn::{Attribute, Expr, File, Item, Local};
+
+pub(crate) fn remove_hidden_attribute(source_file: &mut File) {
+    for item in &mut source_file.items {
+        process_item(item);
+    }
+}
+
+/// Process a syn::Item and its potential nested items
+fn process_item(item: &mut Item) {
```

Comment:
would be much easier to use a visitor pattern for this, see https://docs.rs/syn/latest/syn/visit/index.html and other visitors in this file

Reply:
I had it in the back of my head that doing this nested parsing probably has a proper algorithm somewhere and I should not need to re-invent it 😅 So it is the same visitor alg you have been talking about lately, thanks for sharing

Reply:
In fact, it was my primary area of research in grad school and I've published some novel indexing data structures and algorithms around this problem :)

* https://dickinson.hykucommons.org/concern/etds/97a2026c-684f-4fbc-82eb-6241931a1897?q=AST%20indexing
* https://cs.brown.edu/research/pubs/theses/masters/2017/kelly.samuel.pdf